src/main/content/replays/parse-replay-worker.ts

```diff
@@ -1,11 +1,55 @@
-import { parseReplay } from "@main/content/replays/parse-replay";
-import { isMainThread, parentPort, workerData } from "worker_threads";
+import { DemoParser } from "$/sdfz-demo-parser";
+import { Replay } from "@main/content/replays/replay";
+import path from "path";
+import { isMainThread, parentPort } from "worker_threads";
 
 if (isMainThread) {
     throw new Error("This script should be run in worker thread.");
 } else {
-    const { replayFilePath } = workerData;
-    parseReplay(replayFilePath).then((data) => {
-        parentPort.postMessage(data);
+    const demoParser = new DemoParser();
+    async function parseReplay(replayPath: string) {
+        const replayData = await demoParser.parseDemo(replayPath);
+        const numOfPlayers = replayData.info.players.length + replayData.info.ais.length;
+        let preset: "duel" | "team" | "ffa" | "teamffa" = "duel";
+        if (replayData.info.allyTeams.length > 2 && replayData.info.players.some((player) => player.playerId !== player.allyTeamId)) {
+            preset = "teamffa";
+        } else if (replayData.info.allyTeams.length > 2) {
+            preset = "ffa";
+        } else if (numOfPlayers > 2) {
+            preset = "team";
+        }
+        return {
+            gameId: replayData.header.gameId,
+            fileName: path.parse(replayPath).base,
+            filePath: replayPath,
+            engineVersion: replayData.info.meta.engine,
+            gameVersion: replayData.info.meta.game,
+            mapSpringName: replayData.info.meta.map,
+            startTime: replayData.info.meta.startTime,
+            gameDurationMs: replayData.info.meta.durationMs,
+            gameEndedNormally: replayData.info.meta.winningAllyTeamIds.length > 0 ? 1 : 0,
+            chatlog: replayData.chatlog || null,
+            hasBots: replayData.info.ais.length > 0 ? 1 : 0,
+            preset: preset,
+            winningTeamId: replayData.info.meta.winningAllyTeamIds[0],
+            teams: replayData.info.allyTeams,
+            contenders: [...replayData.info.players, ...replayData.info.ais],
+            spectators: replayData.info.spectators,
+            script: replayData.script,
+            battleSettings: replayData.info.hostSettings,
+            gameSettings: replayData.info.gameSettings,
+            mapSettings: replayData.info.mapSettings,
+            hostSettings: replayData.info.spadsSettings ?? {},
+        } as Replay;
```

Comment:
So that the return type gets inferred by typescript and I get autocompletion and typechecking on the object I'm initializing here. The alternative is to add the return type in the function declaration, that works too.

Reply:
`as` is type casting, you force that object to be of this type regardless if that's true or not.

If you specify it in the function signature, typescript with actually verify if that object matches the type.

Reply:
You are absolutely right, it's a bad practice. Changing this now